api/cluster/ClusterRestHandler.go

```diff
@@ -83,6 +85,70 @@ func NewClusterRestHandlerImpl(clusterService cluster.ClusterService,
 	}
 }
 
+func (impl ClusterRestHandlerImpl) SaveClusters(w http.ResponseWriter, r *http.Request) {
+	token := r.Header.Get("token")
+	decoder := json.NewDecoder(r.Body)
+	userId, err := impl.userService.GetLoggedInUser(r)
+	if userId == 0 || err != nil {
+		common.WriteJsonResp(w, err, "Unauthorized User", http.StatusUnauthorized)
+		return
+	}
+	beans := []cluster.ClusterBean{}
+	err = decoder.Decode(&beans)
+	if err != nil {
+		impl.logger.Errorw("request err, Save", "error", err, "payload", beans)
+		common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
+		return
+	}
+	impl.logger.Infow("request payload, Save", "payload", beans)
+	//err = impl.validator.Struct(beans)
+	//if err != nil {
+	//	impl.logger.Errorw("validation err, Save", "err", err, "payload", beans)
+	//	common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
+	//	return
+	//}
+
+	// RBAC enforcer applying
+	if ok := impl.enforcer.Enforce(token, casbin.ResourceCluster, casbin.ActionCreate, "*"); !ok {
+		common.WriteJsonResp(w, errors.New("unauthorized"), nil, http.StatusForbidden)
+		return
+	}
+	//RBAC enforcer Ends
+	ctx, cancel := context.WithCancel(r.Context())
+	if cn, ok := w.(http.CloseNotifier); ok {
+		go func(done <-chan struct{}, closed <-chan bool) {
+			select {
+			case <-done:
+			case <-closed:
+				cancel()
+			}
+		}(ctx.Done(), cn.CloseNotify())
+	}
+	if util2.IsBaseStack() {
+		ctx = context.WithValue(ctx, "token", token)
+	} else {
+		acdToken, err := impl.argoUserService.GetLatestDevtronArgoCdUserToken()
+		if err != nil {
+			impl.logger.Errorw("error in getting acd token", "err", err)
+			common.WriteJsonResp(w, err, nil, http.StatusInternalServerError)
+			return
+		}
+		ctx = context.WithValue(ctx, "token", acdToken)
+	}
+	result := []cluster.ClusterBean{}
+	for _, bean := range beans {
+		res, err := impl.clusterService.Save(ctx, &bean, userId)
```

Comment:
we should not call our own service within orchestrator with token

Reply:
Ok sure